ps3.py

```diff
@@ -0,0 +1,414 @@
+    """ 
+    Allow the user to replace all copies of one letter in the hand (chosen by user)
+    with a new letter chosen from the VOWELS and CONSONANTS at random. The new letter
+    should be different from user's choice, and should not be any of the letters
+    already in the hand.
+
+    If user provide a letter not in the hand, the hand should be the same.
+
+    Has no side effects: does not mutate hand.
+
+    For example:
+        substitute_hand({'h':1, 'e':1, 'l':2, 'o':1}, 'l')
+    might return:
+        {'h':1, 'e':1, 'o':1, 'x':2} -> if the new letter is 'x'
+    The new letter should not be 'h', 'e', 'l', or 'o' since those letters were
+    already in the hand.
+    
+    hand: dictionary (string -> int)
+    letter: string
+    returns: dictionary (string -> int)
+    """
+    if letter in hand:
+        letters = string.ascii_lowercase
+
+        for i in hand:
+            letters = letters.replace(i, "")
+
+        new_letter = random.choice(letters)
+        hand[new_letter] = hand[letter]
+        del hand[letter]
+
+    return hand
+
+
+def play_game(word_list):
+    """
+    Allow the user to play a series of hands
+
+    * Asks the user to input a total number of hands
+
+    * Accumulates the score for each hand into a total score for the 
+      entire series
+ 
+    * For each hand, before playing, ask the user if they want to substitute
+      one letter for another. If the user inputs 'yes', prompt them for their
+      desired letter. This can only be done once during the game. Once the
+      substitue option is used, the user should not be asked if they want to
+      substitute letters in the future.
+
+    * For each hand, ask the user if they would like to replay the hand.
+      If the user inputs 'yes', they will replay the hand and keep 
+      the better of the two scores for that hand.  This can only be done once 
+      during the game. Once the replay option is used, the user should not
+      be asked if they want to replay future hands. Replaying the hand does
+      not count as one of the total number of hands the user initially
+      wanted to play.
+
+            * Note: if you replay a hand, you do not get the option to substitute
+                    a letter - you must play whatever hand you just had.
+      
+    * Returns the total score for the series of hands
+
+    word_list: list of lowercase strings
+    """
+    game_score = 0
+    substitution = ''
+    replay_hand = ''
+    hand = deal_hand(HAND_SIZE)
+    # Ask user for input
+    hands_number = int(ask_user("Enter total number of hands: ", PATTERNS['hands_number']))
+
+    # This loop works, while amount of hands isn't over
+    while hands_number != 0:
+        print("Current hand:", end=" ")
+        display_hand(hand)
+        print()
+
+        # If the substitution and replay hand aren't used, the program asks user to use substitution
+        if substitution != 'used' and replay_hand != 'yes':
```

Comment:
I thought according to task - it should be possible to substitute a letter on replaying hand, could you check it?

Reply:
```
Note: if you replay a hand, you do not get the option to substitute
                    a letter - you must play whatever hand you just had.
```